app/controllers/articles_controller.rb

```diff
@@ -100,6 +100,23 @@ def redirect
     render "errors/404", :status => 404
   end
 
+  def merge
+    if User.find(session[:user_id]).admin?
+      art = Article.find(params[:id])
+      merge_id = params[:merge_id]
+      merged = art.merge_with(merge_id)
+      if merged
+        redirect_to '/admin/content'
+      else
+        redirect_to :back
+        flash[:error] = "Something went wrong. Please try again."
+      end
+    else
+      redirect_to :back
+      flash[:error] = "You do not have permission to do this."
```

Comment:
Nitpick, but I would prefer seeing the `flash[:error]` prior to the redirect to better explain the flow of the code